### PR TITLE
Build for pypi instead of conda

### PR DESCRIPTION
**Pull request by vladsaveliev**
~Simplify maintaining the conda package recipe: instead of manually changing the list of dependencies, populate them automatically from the pip requirements list.~

Build `cpg-hail` pip package instead of conda.

### CODE REVIEW

File: conda/render_recipe.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        # always reliably resolve:
+        pkg = pkg.split('>')[0].strip()
+        pkg = pkg.split('==')[0].strip()
+        # conda uses dashes where pip uses underscores
+        pkg = pkg.replace('_', '-')
+        if pkg == 'avro':
+            pkg = 'python-avro'
+        dependencies.append(pkg)
+
+with open(META_TEMPLATE) as inp, open(META_RENDERED, 'w') as out:
```
Comment (illusional) [inline conda/render_recipe.py:45]: I don't feel as good about doing this line by line, most templating I've seen uses `.replace('{version}', "<version-string>")` on the whole file, or maybe using a formal templating engine?

**Review by illusional — CHANGES_REQUESTED** [review]
The first half is great, I think templating the conda recipe is fine, but I'm not super comfortable with the script as it stands. The translation of pypi packages to conda packages feels risky, I think it'd be better to completely switch everything over to conda if possible, or I'd feel better about maintaining a json map or something.

I'd prefer a templating engine, or if not I think there should be tests for the templating. Build up the string completely, then write it. There's not much memory benefit doing it line by line.

**Comment by vladsaveliev** [comment]
Modified the workflow to build `cpg-hail` package instead of a conda one.

File: .github/workflows/build_package.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      - name: Redeploy the analysis-server
+        # Only run this step once - that's why limiting to the ubuntu run:
+        if: matrix.os == 'ubuntu-latest' && github.event_name == 'push'
+        run: |
+          pip_version=$(cat hail/python/hail/hail_pip_version)
+          curl --fail --silent --show-error -X POST \
+            -H "Authorization: token ${{ secrets.ANALYSIS_SERVER_GITHUB_TOKEN }}" \
+            -H "Accept: application/vnd.github.v3+json" \
+            https://api.github.com/repos/populationgenomics/analysis-runner/actions/workflows/6364059/dispatches \
+            -d '{"ref": "main", "inputs": {"hail_version": "$pip_version"}}'
```
Comment (illusional) [inline .github/workflows/build_package.yaml:68]: Does the $pip_version resolve correctly, the dictionary is wrapped in single quotes which I didn't think would work.

Comment (illusional) [reply .github/workflows/build_package.yaml:68]: Could replicate the condarise workflow by putting the version into the GITHUB_ENV, then using the GitHub substitution which doesn't rely on the bash substitution:

```
# previous step
echo "version=$VERSION" >> $GITHUB_ENV

# curl [...]
-d '{"ref": "main", "inputs": {"hail_version": "${{ env.version }}"}}'

Comment (vladsaveliev) [reply .github/workflows/build_package.yaml:68]: Good catch!

Comment (vladsaveliev) [reply .github/workflows/build_package.yaml]: Fixed

**Review by illusional — APPROVED** [review]

**Pull request merged**